core/types/block.go

```diff
@@ -1610,3 +1383,167 @@ type BlockWithReceipts struct {
 	Block    *Block
 	Receipts Receipts
 }
+
+// expects either [][]byte or []Transactions, but not both
+func txnsPayloadSize(byteTxns [][]byte, txns []Transaction) (txsLen int) {
```

Comment:
Please split this function in 2 (byte and normal) for clarity

Reply:
Yes, I will. The current problem is that raw transactions are not encoded and decoded correctly. I am having an error with decoding for random inputs larger then 55 bytes. I think this https://github.com/ledgerwatch/erigon/blob/devel/core/types/block.go#L676 should be done a bit differently. Working on it. As soon as I am sure that everything is okay I will make another PR.